lib/subjects/selector.rb

```diff
@@ -128,9 +128,15 @@ def find_subject_queue
     end
 
     def dequeue_ids(queue, sms_ids)
+      background_update = !queue_user
       if queue_user
-        queue.dequeue_update(sms_ids)
-      else
+        begin
+          queue.dequeue_update(sms_ids)
+        rescue ActiveRecord::StaleObjectError
+          background_update = true
```

Comment:
can do for sure, i normally try to extract duplication but agree about readability.